play-wright/e2e/volunteer-registration-form-validation.spec.js

```diff
@@ -0,0 +1,29 @@
+import { test } from '@playwright/test';
+import { userDetails } from '../utils.js';
+import { VolunteerRegistrationPage} from '../pages/volunteer-registration.page.js'
+
+
+test('check mandatory field validation for registration form fields', async ({ page }) => {
+  const volunteerRegistrationPage = new VolunteerRegistrationPage(page);
+  const volunteerUrl = volunteerRegistrationPage.getAppendedUrl('/volunteer-registration/');
+  await page.goto(volunteerUrl);
+  await page.waitForURL(volunteerUrl);
+  await volunteerRegistrationPage.handleDialogMessage('Please fill all required fields.'); // Registering dialog message 
+  await volunteerRegistrationPage.fillAndClearField('enterFirstName', userDetails.firstName);
+  await volunteerRegistrationPage.fillAndClearField('enterLastName', userDetails.lastName);
+  await volunteerRegistrationPage.fillAndClearField('enterEmail', userDetails.email);
+  await volunteerRegistrationPage.fillAndClearField('enterMobileNumber', userDetails.mobileNumber);
+  await volunteerRegistrationPage.selectGenderAndClear(userDetails.gender);
+  await volunteerRegistrationPage.fillAndClearField('enterStreetAddress', userDetails.streetAddress);
+  await volunteerRegistrationPage.fillAndClearField('enterCityName', userDetails.cityName);
+  await volunteerRegistrationPage.fillAndClearField('enterPostalCode', userDetails.postalCode);
+  await volunteerRegistrationPage.selectCountry(userDetails.country) // Selecting country first so that states options can be selected
+  await volunteerRegistrationPage.selectStateAndClear(userDetails.state);
+  await volunteerRegistrationPage.selectCountryAndClear(userDetails.country);
```

Comment:
Will this action change the state dropdown value?

Reply:
@Satendra-SR states are available only when country is selected in country dropdown, hence the code for selecting the country is added in below code 
```suggestion
   await volunteerRegistrationPage.selectCountry(userDetails.country) // Selecting country first so that states options can be selected
  await volunteerRegistrationPage.selectStateAndClear(userDetails.state);
```